pkgs/by-name/go/go-critic/package.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, testers
+, nix-update-script
+, go-critic
+}:
+
+buildGoModule rec {
+  pname = "go-critic";
+  version = "0.11.0";
+
+  src = fetchFromGitHub {
+    owner = "go-critic";
+    repo = "go-critic";
+    rev = "v${version}";
+    hash = "sha256-jL/z1GtHmEbS8vsIYG1jEZOxySXqU92WIq9p+GDTP8E=";
+  };
+
+  vendorHash = "sha256-qQO4JWMU8jfc64CBPaMRYRbUsgLQZx9P5AKbSPyHnRE=";
+
+  subPackages = [
+    "cmd/gocritic"
+  ];
+
+  allowGoReference = true;
+
+  ldflags = [
+    "-X main.Version=${version}"
+  ];
+
+  passthru = {
+    tests.version = testers.testVersion {
+      package = go-critic;
+      command = "gocritic version";
+    };
+    updateScript = nix-update-script { };
```

Comment:
This shouldn't be needed.

Reply:
Why not? Without it, the update script will fail:

```
$ nix-shell ./maintainers/scripts/update.nix --argstr package go-critic
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'nixpkgs-update-script'
         whose name attribute is located at /home/katexochen/katexochen/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'shellHook' of derivation 'nixpkgs-update-script'

         at /home/katexochen/katexochen/nixpkgs/maintainers/scripts/update.nix:223:3:

          222|   '';
          223|   shellHook = ''
             |   ^
          224|     unset shellHook # do not contaminate nested shells

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Package with an attribute name `go-critic` does not have a `passthru.updateScript` attribute defined.
```

Reply:
True, but you can use `nix-update`.

Reply:
I though that `updateScript` was necessary for @r-ryantm to create update PRs 

I know python modules get `updateScript` set to something generic `/nix/store/irasc2pbrj3w69zqdpvdijv39v7j53ln-update-python-libraries.drv`  but for go modules, it seems to be not set.